### PR TITLE
SWC-6286

### PR DESCRIPTION
**Pull request by nickgros**
ApproveUserAccessModal: Show error message and hide dialog when no compatible ARs are found

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/widget/entity/act/ApproveUserAccessModal.java [hunk 1]
```diff
@@ -48,48 +49,47 @@ public class ApproveUserAccessModal implements ApproveUserAccessModalView.Presen
 	public static final String REVOKED_USER = "Successfully Revoked User Access. ";
 	public static final String EMAIL_SENT = "An email has been sent to notify them";
 	public static final String MESSAGE_BLANK = "You must enter an email message to approve this user";
-	public static final String NO_APPROVAL_FOUND = "There was no approval found for the specified user and requirement";
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/act/ApproveUserAccessModal.java:51]: Unused

File: src/main/java/org/sagebionetworks/web/client/widget/entity/act/ApproveUserAccessModal.java [hunk 2]
```diff
@@ -128,12 +124,15 @@ public void configure(List<AccessRequirement> accessRequirements, EntityBundle b
...
+			view.setDatasetTitle(bundle.getEntity().getName());
+			loadEmailMessage();
+		} else {
+			popupUtils.showErrorMessage(NO_COMPATIBLE_ARS_MESSAGE);
+			view.hide();
 		}
-		datasetId = entityBundle.getEntity().getId(); // get synId of dataset we are currently on
-		view.setDatasetTitle(entityBundle.getEntity().getName());
-		loadEmailMessage();
 	}
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/act/ApproveUserAccessModal.java:134]: This is the only logical change

File: src/main/java/org/sagebionetworks/web/client/widget/entity/act/ApproveUserAccessModal.java [hunk 3]
```diff
@@ -19,6 +19,7 @@
 import org.sagebionetworks.repo.model.table.Row;
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/act/ApproveUserAccessModal.java:19]: Made a bunch of IDE-suggested "clean up" changes. Only logical change is pointed out below.

File: src/main/java/org/sagebionetworks/web/client/widget/entity/act/ApproveUserAccessModal.java [hunk 4]
```diff
@@ -48,48 +49,47 @@ public class ApproveUserAccessModal implements ApproveUserAccessModalView.Presen
 	public static final String REVOKED_USER = "Successfully Revoked User Access. ";
 	public static final String EMAIL_SENT = "An email has been sent to notify them";
 	public static final String MESSAGE_BLANK = "You must enter an email message to approve this user";
-	public static final String NO_APPROVAL_FOUND = "There was no approval found for the specified user and requirement";
+
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/act/ApproveUserAccessModal.java]:
```suggestion
```

File: src/main/java/org/sagebionetworks/web/client/widget/entity/act/ApproveUserAccessModal.java [hunk 5]
```diff
@@ -48,48 +49,47 @@ public class ApproveUserAccessModal implements ApproveUserAccessModalView.Presen
...
-	private RestrictableObjectDescriptor subject;
+	private final ApproveUserAccessModalView view;
+	private final SynapseAlert synAlert;
+	private final SynapseClientAsync synapseClient;
+	private final SynapseProperties synapseProperties;
+	private final JobTrackingWidget progressWidget;
+	private final DataAccessClientAsync dataAccessClient;
+	private final AccessRequirementWidget arWidget;
+	private final RestrictableObjectDescriptor subject;
+
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/act/ApproveUserAccessModal.java]:
```suggestion
```

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**